packages/in_app_purchase/example/android/app/src/test/java/io/flutter/plugins/inapppurchase/InAppPurchasePluginTest.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.inapppurchase;
+
+import static org.mockito.Mockito.when;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.PluginRegistry;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class InAppPurchasePluginTest {
+  @Mock Activity activity;
+  @Mock Context context;
+  @Mock PluginRegistry.Registrar mockRegistrar; // For v1 embedding
+  @Mock BinaryMessenger mockMessenger;
+  @Mock Application mockApplication;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    when(mockRegistrar.activity()).thenReturn(activity);
+    when(mockRegistrar.messenger()).thenReturn(mockMessenger);
+    when(mockRegistrar.context()).thenReturn(context);
+  }
+
+  @Test
+  public void registerWith_doNotCrashWhenApplicationContextIsTypeApplication() {
```

Comment:
Is this test necessary? We should probably avoid testing situations that can't happen, because it might indicate to readers that it can happen...

Reply:
Isn't `context.getApplicationContext()` always going to return an application? This looks like the standard case to me.

Reply:
Though on a second look, maybe a more meaningful test here would be to return an `Activity` from `registrar.context()` and make sure that that doesn't crash?

Reply:
@mklim Sounds good, I will update the test.